src/Microsoft.Extensions.Options/OptionsServiceCollectionExtensions.cs

```diff
@@ -50,8 +63,11 @@ public static IServiceCollection Configure<TOptions>(this IServiceCollection ser
                 throw new ArgumentNullException(nameof(configureOptions));
             }
 
-            services.AddSingleton<IConfigureOptions<TOptions>>(new ConfigureOptions<TOptions>(configureOptions));
+            services.AddSingleton<IConfigureOptions<TOptions>>(new ConfigureOptions<TOptions>(name, configureOptions));
             return services;
         }
+
+        public static IServiceCollection ConfigureAll<TOptions>(this IServiceCollection services, Action<TOptions> configureOptions) where TOptions : class
```

Comment:
It should just result in a new instance of ConfigureOptions { Name = null, action = action }.  Which then eliminates the name check in Configure via `(if Name == null || name == Name)`